src/compiler/checker.ts

```diff
@@ -4542,7 +4542,7 @@ namespace ts {
             return type;
         }
 
-        function getTypeFromStringLiteral(node: StringLiteral): Type {
+        function getTypeFromStringLiteral(node: StringLiteral | StringLiteralTypeNode): Type {

```

Comment:
Can you add a `// TODO (drosen): Use this function in 'checkStringLiteralExpression'.`?

I need to see how this affects things, but I don't want it to affect your work.